Add compose ui for channel page (version 1)

AmyPlace opened the pull request:
- 先上第一版的 channel 畫面
- 未完成
-- 收合 toolbar
-- sticky header (所有章節)
-- channel 介紹資訊的展開與收合

AmyPlace committed:
Add compose ui for channel page (version 1)

HsienDaDa commented on app/src/main/java/tw/jizah/popocast/ui/channel/Channel.kt:
```diff
@@ -0,0 +1,213 @@
+@Composable
+fun ChannelPage(channelInfo: ChannelInfo) {
+
+    Column(modifier = Modifier.fillMaxSize().background(Color.Black)) {
```
Colors.black

AmyPlace replied:
done

HsienDaDa commented on app/src/main/java/tw/jizah/popocast/ui/channel/Channel.kt:
```diff
@@ -0,0 +1,213 @@
+                }) {
+                Icon(
+                    asset = Icons.Filled.ArrowBack,
+                    tint = Color.White
```
Colors.white

AmyPlace replied:
done

HsienDaDa commented on app/src/main/java/tw/jizah/popocast/ui/channel/Channel.kt:
```diff
@@ -0,0 +1,213 @@
+                fontWeight = FontWeight.Bold,
+                modifier = Modifier.constrainAs(titleId) {
+                    start.linkTo(coverId.end, m4)
+                    end.linkTo(parent.end)
```
parent是不是也有right?
若不考慮RTL的話是不是用 parent.right?

AmyPlace replied:
決議不使用 left/right

ivanisidrowu commented on app/src/main/java/tw/jizah/popocast/ui/channel/Channel.kt:
```diff
@@ -0,0 +1,213 @@
+import java.util.*
+import java.util.concurrent.TimeUnit
+
+data class ChannelInfo(
```
這個要不要獨立一個 class 會比較好？

AmyPlace replied:
done

dev-weiqi commented on app/src/main/java/tw/jizah/popocast/ui/channel/Channel.kt:
```diff
@@ -0,0 +1,213 @@
+                colors = ButtonConstants.defaultButtonColors(
+                    backgroundColor = Color.Black
+                ),
+                border = BorderStroke(1.dp, Color.White),
```
1dp 要寫在dimens嗎？

AmyPlace replied:
done

dev-weiqi commented on app/src/main/java/tw/jizah/popocast/ui/channel/Item.kt:
```diff
@@ -0,0 +1,268 @@
+            EllipsisText(text = itemInfo,
+                maxLines = 2,
+                style = MaterialTheme.typography.subtitle2,
+                color = Color.LightGray,
```
好像滿多color沒寫在colors裡面

AmyPlace replied:
done

dev-weiqi reviewed (commented):
Done

ivanisidrowu commented on app/src/main/java/tw/jizah/popocast/ui/channel/Item.kt:
```diff
@@ -0,0 +1,268 @@
+    Card(
+        modifier = modifier,
+        shape = RoundedCornerShape(m2),
+        backgroundColor = Color.DarkGray
```
可以使用 Colors 裡面的

ivanisidrowu replied:
下面的顏色也有類似的狀況，所以我就不留言了

AmyPlace replied:
done

ivanisidrowu commented on app/src/main/res/values/strings.xml:
```diff
@@ -1,3 +1,9 @@
 <resources>
     <string name="app_name">PoPoCast</string>
+    <string name="all_chapter">所有章節</string>
```
我們是統一寫英文還是統一寫中文？  還是要先做 localization?

AmyPlace replied:
決議先支援英文跟繁中

AmyPlace committed:
1. Add English/Traditional Chinese localization
2. define colors in Colors.kt
3. define custom dp in Dimens.kt
4. add files to model package

ivanisidrowu reviewed (approved):
🤘🏼

AmyPlace committed:
Merge remote-tracking branch 'origin/main' into channel

# Conflicts:
#	app/src/main/java/tw/jizah/popocast/ui/theme/Colors.kt
#	app/src/main/java/tw/jizah/popocast/ui/theme/Dimens.kt
#	app/src/main/res/values/strings.xml

AmyPlace committed:
1. add traditional chinese strings
2. Use Colors.black instead of black
3. Use Dimens.mx instead of mx
